src/main/resources/io/viash/runners/nextflow/workflowFactory/workflowFactory.nf

```diff
@@ -185,44 +189,95 @@ def workflowFactory(Map args, Map defaultWfArgs, Map meta) {
         output_ = output_.findAll{k, v -> k != "_meta"}
 
         // check value types
-        output_ = _processOutputValues(output_, meta.config, id_, key_)
-
-        // simplify output if need be
-        if (workflowArgs.auto.simplifyOutput && output_.size() == 1) {
-          output_ = output_.values()[0]
-        }
+        output_ = _checkValidOutputArgument(output_, meta.config, id_, key_)
 
         [join_id, id_, output_]
       }
       // | view{"chInitialOutput: ${it.take(3)}"}
 
     // join the output [prev_id, new_id, output] with the previous state [prev_id, state, ...]
-    def chNewState = safeJoin(chInitialOutput, chRunFiltered, key_)
+    def chPublishWithPreviousState = safeJoin(chInitialOutputProcessed, chRunFiltered, key_)
       // input tuple format: [join_id, id, output, prev_state, ...]
       // output tuple format: [join_id, id, new_state, ...]
       | map{ tup ->
         def new_state = workflowArgs.toState(tup.drop(1).take(3))
         tup.take(2) + [new_state] + tup.drop(4)
       }
-
     if (workflowArgs.auto.publish == "state") {
```

Comment:
Currently, this mimicks the old behavior: "state" means that both the state.yaml and the files are published. This PR separates the publishing of the state and files, but is fully backwards compatible. Changes based on other values of `workflowArgs.auto.publish` can be introduced with other PRs